examples/preprocessing/plot_all_scaling.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+=============================================================
+Compare the effect of different scalers on data with outliers
+=============================================================
+
+Feature 0 (median income in a block) and feature 5 (number of households) of
+the `California housing dataset
+<http://www.dcc.fc.up.pt/~ltorgo/Regression/cal_housing.html>`_ have very
+different scales and contain some very large outliers. These two
+characteristics lead to difficulties to visualize the data and, more
+importantly, they can degrade the predictive performance of many machine
+learning algorithms. Unscaled data can also slow down or even prevent the
+convergence of many gradient-based estimators.
+
+Indeed many estimators are designed with the assumption that each feature takes
+values close to zero or more importantly that all features vary on comparable
+scales. In particular, metric-based and gradient-based estimators often assume
+approximately standardized data (centered features with unit variances). A
+notable exception are decision tree-based estimators that are robust to
+arbitrary scaling of the data.
+
+This example uses different scalers, transformers, and normalizers to bring the
+data within a pre-defined range.
+
+Scalers are linear (or more precisely affine) transformers and differ from each
+other in the way to estimate the parameters used to shift and scale each
+feature.
+
+``QuantileTransformer`` provides a non-linear transformation in which distances
+between marginal outliers and inliers are shrunk.
+
+Unlike the previous transformations, normalization refers to a per sample
+transformation instead of a per feature transformation.
+
+The following code is a bit verbose, feel free to jump directly to the analysis
+of the results_.
+
+"""
+
+# Author:  Raghav RV <rvraghav93@gmail.com>
+#          Guillaume Lemaitre <g.lemaitre58@gmail.com>
+#          Thomas Unterthiner
+# License: BSD 3 clause
+
+from __future__ import print_function
+
+from collections import OrderedDict
+
+import numpy as np
+
+import matplotlib as mpl
+from matplotlib import pyplot as plt
+from matplotlib import cm
+
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.preprocessing import minmax_scale
+from sklearn.preprocessing import MaxAbsScaler
+from sklearn.preprocessing import StandardScaler
+from sklearn.preprocessing import RobustScaler
+from sklearn.preprocessing import Normalizer
+from sklearn.preprocessing.data import QuantileTransformer
+
+from sklearn.datasets import fetch_california_housing
+
+print(__doc__)
+
+dataset = fetch_california_housing()
+X_full, y_full = dataset.data, dataset.target
+
+# Take only 2 features to make visualization easier
+# Feature of 0 has a long tail distribution.
+# Feature 5 has a few but very large outliers.
+
+X = X_full[:, [0, 5]]
+
+distributions = OrderedDict((
```

Comment:
It seems the OrderedDict is quite an overkill as you use positional indexers anyways. I'd suggest to just use a list of tuples.